djstripe/models.py

```diff
@@ -593,6 +563,39 @@ def record_charge(self, charge_id):
         data = stripe.Charge.retrieve(charge_id)
         return Charge.sync_from_stripe_data(data)
 
+    @staticmethod
+    @webhook.handler_all
+    def event_attach_customer(event, data, cat, sub_cat):
+        stripe_customer_crud_events = ["created", "updated", "deleted", ]
+        skip_events = ["plan", "transfer", ]
+
+        if cat in skip_events:
+            return
+        elif cat == "customer" and sub_cat in stripe_customer_crud_events:
+            stripe_customer_id = data["object"]["id"]
+        else:
+            stripe_customer_id = data["object"].get("customer", None)
+
+        if stripe_customer_id:
+            try:
+                event.customer = Customer.objects.get(stripe_id=stripe_customer_id)
```

Comment:
I dropped the extra save() that used to be there. My thinking was: any event handler should be allowed to modify the event data (attaching objects to the event, etc) and why not delay the save() until all event processing is done? Reduce signalling, reduce db calls, also decouple the question of whether or not we want to save the event at all... 

But yes, I did drop a save(). That was the only spot where I made a semantic change (at least as far as I know) in this refactor.